pandas/core/groupby.py

```diff
@@ -2553,9 +2595,17 @@ def group_index(self):
 
     def _make_labels(self):
         if self._labels is None or self._group_index is None:
-            labels, uniques = algorithms.factorize(
-                self.grouper, sort=self.sort)
-            uniques = Index(uniques, name=self.name)
+            # for the situation of groupby list of groupers
+            if isinstance(self.grouper, BaseGrouper):
+                labels, _, _ = self.grouper.group_info
+                uniques = self.grouper.result_index
+                if self.grouper.indexer is not None:
```

Comment:
refactor to make a method on the BaseGrouper itself and override in BinGrouper; same for indices

Reply:
I'm afraid cannot use this way. Grouping use the group_info for unsorted axis, while some other scenarios call the group_info to group sorted axis, I think it's better to keep the group_info as sorted, and reorder to get the unsorted label sequence.

Reply:
that's fine, but not my point. I don't want these if/else in the properties, rather they should simply be overriden methods on the grouper type

Reply:
OK I get your point.